pkg/mutation/mutators/assignimage/errors.go

```diff
@@ -27,23 +27,23 @@ type metadataRootError struct{ baseError }
 func newInvalidDomainError(domain string) invalidDomainError {
 	return invalidDomainError{
 		baseError{
-			fmt.Sprintf("assignDomain %q does not match pattern %s", domain, domainRegexp.String()),
+			fmt.Sprintf("assignDomain %q must be a fully-qualified domain name or localhost", domain),
 		},
 	}
 }
 
 func newInvalidPathError(path string) invalidPathError {
 	return invalidPathError{
 		baseError{
-			fmt.Sprintf("assignPath %q does not match pattern %s", path, pathRegexp.String()),
+			fmt.Sprintf("assignPath %q must be a valid docker image path", path),
 		},
 	}
 }
 
 func newInvalidTagError(tag string) invalidTagError {
 	return invalidTagError{
 		baseError{
-			fmt.Sprintf("assignTag %q does not match pattern %s", tag, tagRegexp.String()),
+			fmt.Sprintf("assignTag %q must be a valid docker image tag", tag),
```

Comment:
nit: tag or hash

Reply:
Good catch. Technically, I think `digest` is the [terminology docker](https://docs.docker.com/engine/reference/commandline/images/#-list-image-digests---digests) uses for this though.

Also, since we are not exposing the regex anymore, I am a little concerned it may not be obvious to the user that the tag must start with `:` or `@`. In accordance, I updated the error to say:
```
assignTag %q must be a valid docker image tag or digest starting with ':' or '@'
```

WDYT?